zeebe/backup/src/main/java/io/camunda/zeebe/backup/api/BackupIdentifierWildcard.java

```diff
@@ -27,12 +30,128 @@ public interface BackupIdentifierWildcard {
   /**
    * @return id of the checkpoint included in the backup
    */
-  Optional<Long> checkpointId();
+  CheckpointPattern checkpointPattern();
 
   /**
    * Predicate that tries to match an id to this wildcard.
    *
    * @return true if the given id matches the wildcard, otherwise false.
    */
   boolean matches(BackupIdentifier id);
+
+  /**
+   * Tries to build the longest possible prefix based on the given wildcard. The prefix is
+   * constructed as follows: {@code ${partitionId}/${checkpointId}/${nodeId}}. If a field is not
+   * present or is a non-exact match, all following fields are omitted. For example, if the
+   * checkpoint id should match a {@link CheckpointPattern.Prefix prefix}, the returned prefix is
+   * {@code ${partitionId}/${checkpointPattern.Prefix}}. If none of the fields are present, an empty
+   * string is returned.
+   */
+  static String asPrefix(final BackupIdentifierWildcard wildcard) {
+    final var prefix = new StringBuilder();
+    if (wildcard.partitionId().isEmpty()) {
+      return prefix.toString();
+    }
+    prefix.append(wildcard.partitionId().get());
+    prefix.append("/");
+
+    switch (wildcard.checkpointPattern()) {
+      case Exact(final var checkpointId) -> {
+        prefix.append(checkpointId);
+        prefix.append("/");
+        // Checkpoint pattern is exact so we can include node id if present
+        if (wildcard.nodeId().isPresent()) {
+          prefix.append(wildcard.nodeId().get());
+        }
+      }
+      // Checkpoint pattern is not exact so our prefix ends here, we can't include the node id
+      case Prefix(final var cpPrefix) -> prefix.append(cpPrefix);
+      case Any() -> {}
```

Comment:
💭 Took me a while to understand this. Maybe it is just me, but I feel this could be easier/reduced 🤔

Reply:
Imagine we add support for matching checkpoints by regex. I think it's way easier to just add one record and then get a compile error telling me where I need to add support, rather than somehow detecting it's a regex by the contents of the string and then going over all usages manually, hoping I don't forget any.